src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,402 @@
+                        "is out of the range of existing indexes!");
+            }
+        } else {
+            throw new DukeException();
+        }
+    }
+
+    public static void deleteCommand(String[] command, ArrayList<Task> tasks) throws DukeException{
+        if(command.length == 2 && checkIfInteger(command[1])){
+            int index = Integer.parseInt(command[1]) - 1;
+            if (0 <= index && index < tasks.size()) {
+                // If the given value to delete is an existing index, allow deletion
+                deleteTaskMessage(tasks, index);
+                tasks.remove(index);
+            } else {
+                System.out.println("The input index that you have selected to delete, "+
+                        "is out of the range of existing indexes!");
+            }
+        } else {
+            throw new DukeException();
+        }
+    }
+
+    public static Task todoCommand(String[] command, ArrayList<Task> tasks) throws DukeException{
+        if(command.length > 1) {
+            Task newItem = new Todo(String.join(" ", Arrays.copyOfRange(command, 1, command.length)));
+            tasks.add(newItem);
+            return newItem;
+        } else {
+            throw new DukeException();
+        }
+    }
+
+    public static Task eventCommand(String[] command, ArrayList<Task> tasks) throws DukeException{
+        if(checkForSubstring(command, "/at")){
+            int separatorIndex = indexOfSubstring(command, "/at");
+            String description = String.join(" ", Arrays.copyOfRange(command,
+                    1, separatorIndex));
+            String at = String.join(" ",Arrays.copyOfRange(command,
+                    separatorIndex + 1, command.length));
+
+            Task newItem = new Event(description, at);
+            tasks.add(newItem);
+            return newItem;
+        } else{
+            throw new DukeException();
+        }
+    }
+
+    public static Task deadlineCommand(String[] command, ArrayList<Task> tasks) throws DukeException{
+        if(checkForSubstring(command, "/by")){
+            int separatorIndex = indexOfSubstring(command, "/by");
+            String description = String.join(" ", Arrays.copyOfRange(command,
+                    1, separatorIndex));
+            String by = String.join(" ",Arrays.copyOfRange(command,
+                    separatorIndex + 1, command.length));
+
+            Task newItem = new Deadline(description, by);
+            tasks.add(newItem);
+            return newItem;
+        } else{
+            throw new DukeException();
+        }
+    }
+
+    public static boolean checkForSubstring(String[] input, String substring){
+        for(String string : input){
+            if(string.equals(substring)){
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static int indexOfSubstring(String[] input, String substring){
```

Comment:
The name of the method is probably better to be an action. So maybe "getIndexOfSubString"?